src/Illuminate/Foundation/Console/KeyGenerateCommand.php

```diff
@@ -111,6 +119,40 @@ protected function writeNewEnvironmentFileWith($key)
         return true;
     }
 
+    /**
+     * Move the current APP_KEY to APP_PREVIOUS_KEYS.
+     *
+     * @param  string  $envContent
+     * @param  string  $currentKey
+     * @return string
+     */
+    protected function moveToPreviousKeys($envContent, $currentKey)
+    {
+        preg_match('/^APP_PREVIOUS_KEYS=(.*)$/m', $envContent, $matches);
+
+        $existingPreviousKeys = isset($matches[1]) ? trim($matches[1]) : '';
+
+        $previousKeysArray = array_filter(explode(',', $existingPreviousKeys));
+        $previousKeysArray[] = $currentKey;
+        $newPreviousKeys = implode(',', $previousKeysArray);
+
+        if (preg_match('/^APP_PREVIOUS_KEYS=/m', $envContent)) {
+            $envContent = preg_replace(
+                '/^APP_PREVIOUS_KEYS=.*$/m',
+                'APP_PREVIOUS_KEYS='.$newPreviousKeys,
+                $envContent
+            );
+        } else {
+            $envContent = preg_replace(
+                '/^(APP_KEY=.*$)/m',
+                "$1\nAPP_PREVIOUS_KEYS=".$newPreviousKeys,
```

Comment:
Here as well, you could reuse existing code

```suggestion
                $this->keyReplacementPattern(),
                "$0\nAPP_PREVIOUS_KEYS=".$newPreviousKeys,
```